libsolidity/analysis/TypeChecker.cpp

```diff
@@ -81,11 +82,11 @@ bool TypeChecker::visit(ContractDefinition const& _contract)
 	if (function)
 	{
 		if (!function->returnParameters().empty())
-			typeError(function->returnParameterList()->location(), "Non-empty \"returns\" directive for constructor.");
+			m_errorReporter.fatalTypeError(function->returnParameterList()->location(), "Non-empty \"returns\" directive for constructor.");
```

Comment:
Most of these aren't fatal type errors, just regular ones. Can you please review?